presto-memsql/src/test/java/io/prestosql/plugin/memsql/TestMemSqlTypeMapping.java

```diff
@@ -0,0 +1,341 @@
+import static io.prestosql.testing.datatype.DataType.stringDataType;
+import static io.prestosql.testing.datatype.DataType.tinyintDataType;
+import static io.prestosql.testing.datatype.DataType.varcharDataType;
+import static java.lang.String.format;
+
+public class TestMemSqlTypeMapping
+        extends AbstractTestQueryFramework
+{
+    private static final String CHARACTER_SET_UTF8 = "CHARACTER SET utf8";
+
+    protected TestingMemSqlServer memSqlServer;
+
+    @Override
+    protected QueryRunner createQueryRunner()
+            throws Exception
+    {
+        memSqlServer = new TestingMemSqlServer();
+        return createMemSqlQueryRunner(memSqlServer);
+    }
+
+    @AfterClass(alwaysRun = true)
+    public final void destroy()
+    {
+        memSqlServer.close();
+    }
+
+    @Test
+    public void testBasicTypes()
+    {
+        DataTypeTest.create()
+                .addRoundTrip(bigintDataType(), 123_456_789_012L)
+                .addRoundTrip(integerDataType(), 1_234_567_890)
+                .addRoundTrip(smallintDataType(), (short) 32_456)
+                .addRoundTrip(tinyintDataType(), (byte) 125)
+                .addRoundTrip(doubleDataType(), 123.45d)
+                // TODO (https://github.com/prestosql/presto/issues/5452) .addRoundTrip(realDataType(), 123.45f)
+                .execute(getQueryRunner(), prestoCreateAsSelect("test_basic_types"));
+    }
+
+    @Test
+    public void testFloat()
+    {
+        // TODO (https://github.com/prestosql/presto/issues/5452)
+//        singlePrecisionFloatingPointTests(realDataType())
+//                .execute(getQueryRunner(), prestoCreateAsSelect("presto_test_float"));
+        singlePrecisionFloatingPointTests(memSqlFloatDataType())
+                .execute(getQueryRunner(), memSqlCreateAndInsert("tpch.memsql_test_float"));
+    }
+
+    private static DataTypeTest singlePrecisionFloatingPointTests(DataType<Float> floatType)
+    {
+        // we are not testing Nan/-Infinity/+Infinity as those are not supported by MemSQL
+        return DataTypeTest.create()
+                .addRoundTrip(floatType, 3.14f)
+                // .addRoundTrip(floatType, 3.1415927f) // Overeagerly rounded by MemSQL to 3.14159 TODO
+                .addRoundTrip(floatType, null);
+    }
+
+    @Test
+    public void testDouble()
+    {
+        doublePrecisionFloatingPointTests(doubleDataType())
+                .execute(getQueryRunner(), prestoCreateAsSelect("presto_test_double"));
+        doublePrecisionFloatingPointTests(memSqlDoubleDataType())
+                .execute(getQueryRunner(), memSqlCreateAndInsert("tpch.memsql_test_double"));
+    }
+
+    private static DataTypeTest doublePrecisionFloatingPointTests(DataType<Double> doubleType)
+    {
+        // we are not testing Nan/-Infinity/+Infinity as those are not supported by MemSQL
+        return DataTypeTest.create()
+                .addRoundTrip(doubleType, 1.0e100d)
```

Comment:
Same comment from https://github.com/prestosql/presto/pull/5455/files#diff-52d03d3dd03c586bef40fbfe81adcc77R98 should apply here too?

Reply:
there is only one significant digit here, no precision truncation

(btw the value is interesting because its way outside of 64-bit integer range)